homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,216 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from mypermobil import (
+    BATTERY_AMPERE_HOURS_LEFT,
+    BATTERY_CHARGE_TIME_LEFT,
+    BATTERY_DISTANCE_LEFT,
+    BATTERY_INDOOR_DRIVE_TIME,
+    BATTERY_MAX_AMPERE_HOURS,
+    BATTERY_MAX_DISTANCE_LEFT,
+    BATTERY_STATE_OF_CHARGE,
+    BATTERY_STATE_OF_HEALTH,
+    RECORDS_SEATING,
+    USAGE_ADJUSTMENTS,
+    USAGE_DISTANCE,
+)
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import PERCENTAGE, UnitOfEnergy, UnitOfLength, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import BATTERY_ASSUMED_VOLTAGE, DOMAIN
+from .coordinator import MyPermobilCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PermobilRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Any], Any]
```

Comment:
```suggestion
    value_fn: Callable[[Any], float | None]
```

You specified `float | None` as return type of the native value in https://github.com/home-assistant/core/pull/95613/files#diff-0bbb1250ed1272a89074368fdeeab9775aa930823136e060d3273d5b1f39853fR211